starwars/src/App.js

```diff
@@ -1,19 +1,26 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 import './App.css';
+import { CharacterList } from './components/CharacterList';
 
 const App = () => {
-  // Try to think through what state you'll need for this app before starting. Then build out
-  // the state properties here.
+  const [characters, setCharacters] = useState([]);
 
-  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
-  // side effect in a component, you want to think about which state and/or props it should
-  // sync up with, if any.
+  useEffect(() => {
+    axios
+    .get(`https://swapi.co/api/people/`)
+    .then(res => {
+      setCharacters(res.data.results);
+    })
+    .catch(err => console.log(err));
```

Comment:
A better way to handle error is to create another slice of state, then render the error in the view
e.g
```
const [error, setError] = useState(null);
```
then in the catch block
```
 .catch(error => {
        setError(error.message);
      })
```
then in the jsx:
```
{ error && <p>{error}</p>